core/server/common/src/main/java/alluxio/underfs/UfsManager.java

```diff
@@ -20,17 +21,46 @@
  * A class that manages the UFS used by different services.
  */
 public interface UfsManager extends Closeable {
+  /** Container for a UFS and the URI for that UFS. */
+  static class Ufs {
```

Comment:
Which parts of the code base should interact with `Ufs` and which parts should interact with `UnderFileSystem`? Can we always pass `Ufs` and only access `UnderFileSystem` when we need to make a call? Alternatively, do you think it is possible to bundle all the metadata into the UfsConfiguration?